### ISSUE TITLE
conda is unable to find a solution for a long time and is killed by RAM watchdog

### ISSUE THREAD

**Issue by adalisan** [conda/conda#8951]
## Current Behavior

conda install -v -v -c pytorch faiss-gpu 
runs for at least 15 minutes and is killed by RAM watchdog (conda uses  more than 5G, I think.)
verbose output repeats this  many times (most likely an infinite loop)
DEBUG conda.models.prefix_graph:_topo_sort_handle_cycles(259): CyclicalDependencyError: Cyclic dependencies exist among these items:
  - defaults::opencv-3.3.1-py36h9bb06ff_1
  - defaults::shapely-1.6.4-py36h0c48222_0
  - defaults::opencv-3.3.1-py27hdcf4849_0
  - defaults::py-opencv-3.4.2-py27h8fa1ad8_0
  - defaults::py-opencv-3.4.2-py27he6673a0_0
  - defaults::opencv-3.3.1-py27h6cbbc71_1
  - defaults::scipy-1.2.1-py37he2b7bc3_0
  - defaults::py-opencv-3.4.2-py27h765d7f9_1
  - defaults::opencv-3.3.1-py36h6cbbc71_1
  - defaults::py-opencv-3.4.2-py37he6673a0_0
  - defaults::shapely-1.6.4-py36h7ef4460_0
  - conda-forge::scikit-image-0.15.0-py36hb3f55d8_2
  - defaults::py-opencv-3.4.2-py37h765d7f9_1
  - defaults::opencv-3.3.1-py36h0f6f1c3_0
  - defaults::opencv-3.3.1-py27h9bb06ff_1
  - defaults::pywavelets-1.0.2-py36hdd07704_0
  - defaults::py-opencv-3.4.2-py37h8fa1ad8_0
  - defaults::scipy-1.2.1-py37h7c811a0_0
  - defaults::py-opencv-3.4.2-py27hb342d67_1
  - defaults::scikit-image-0.14.2-py27he6710b0_0
  - defaults::numpy-1.16.4-py37h7e9f1db_0
  - defaults::pywavelets-1.0.3-py37hdd07704_1
  - conda-forge::imgaug-0.2.9-py_0
  - conda-forge::scikit-image-0.15.0-py37hb3f55d8_2
  - defaults::py-opencv-3.4.2-py37hb342d67_1
  - conda-forge::scikit-image-0.15.0-py37he1b5a44_0
  - defaults::mkl_fft-1.0.12-py37ha843d7b_0
  - defaults::pywavelets-1.0.2-py27hdd07704_0
  - conda-forge::scikit-image-0.15.0-py36he1b5a44_0
  - defaults::scikit-image-0.14.2-py36he6710b0_0
  - defaults::shapely-1.6.4-py27h0c48222_0
  - defaults::py-opencv-3.4.2-py36he6673a0_0
  - defaults::scipy-1.2.1-py27h7c811a0_0
  - defaults::shapely-1.6.4-py37h7ef4460_0
  - defaults::scipy-1.2.1-py36h7c811a0_0
  - defaults::scikit-image-0.15.0-py36he6710b0_0
  - defaults::shapely-1.6.4-py36h86c5351_0
  - defaults::pywavelets-1.0.2-py37hdd07704_0
  - defaults::scipy-1.2.1-py27he2b7bc3_0
  - defaults::scikit-image-0.14.2-py37he6710b0_0
  - defaults::opencv-3.3.1-py27h17fce66_0
  - defaults::shapely-1.6.4-py37h86c5351_0
  - defaults::scipy-1.2.1-py36he2b7bc3_0
  - defaults::py-opencv-3.4.2-py36h8fa1ad8_0
  - conda-forge::imgaug-0.2.9-py_1
  - conda-forge::albumentations-0.3.0-py_0
  - defaults::opencv-3.3.1-py36h0a11808_0
  - defaults::shapely-1.6.4-py27h86c5351_0
  - defaults::scikit-image-0.15.0-py37he6710b0_0
  - defaults::py-opencv-3.4.2-py36hb342d67_1
  - defaults::pywavelets-1.0.3-py36hdd07704_1
  - defaults::py-opencv-3.4.2-py36h765d7f9_1
  - defaults::shapely-1.6.4-py27h7ef4460_0

### Steps to Reproduce

```
conda install -v -v -c pytorch faiss-gpu 
```

## Expected Behavior

should install the package

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : pytorch_gpu
    active env location : <removed>/anaconda3/envs/pytorch_gpu
            shell level : 1
       user config file : /home/sadali/.condarc
 populated config files : /home/sadali/.condarc
          conda version : 4.7.7
    conda-build version : 3.18.7
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.0
       base environment : <removed>  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : <removed>/anaconda3/pkgs
                          /home/sadali/.conda/pkgs
       envs directories : <removed>/anaconda3/envs
                          /home/sadali/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.7 requests/2.22.0 CPython/3.7.3 Linux/3.10.0-862.2.3.el7.x86_64 centos/7.5.1804 glibc/2.17
                UID:GID : 8513:34
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /home/sadali/.condarc <==
auto_activate_base: False
safety_checks: disabled
ssl_verify: False
channels:
  - defaults
  - conda-forge
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```

```
</p></details>

**Comment by msarahan** [comment]
This should be fixed in 4.7.8, which is on the conda-canary and conda-forge channels right now.

**Comment by johndavidmiller** [comment]
Merde! I'm having this issue on 4.7.10, installing mlflow (see below).  With a fresh install and only one addition conda install (with several packages at once)

Why on earth is it worrying about multiple old versions, and for multiple versions of Python - including 2.7?  

How do I get past this?



DEBUG conda.models.prefix_graph:_topo_sort_handle_cycles(259): CyclicalDependencyError: Cyclic dependencies exist among these items: 
  - defaults::numpy-1.15.4-py37h99e49ec_0
  - defaults::numpy-1.11.3-py37h99e49ec_11
  - conda-forge::numpy-1.14.6-py37_blas_openblashb06ca3d_0
  - conda-forge::numpy-1.15.2-py37_blas_openblashd3ea46f_1
  - conda-forge::numpy-1.15.2-py37_blas_openblashb06ca3d_1
  - defaults::numpy-1.15.2-py37h99e49ec_1
  - conda-forge::matplotlib-3.1.0-py37_1
  - conda-forge::matplotlib-3.1.1-py37_0
  - defaults::matplotlib-3.0.3-py37h5429711_0
  - defaults::matplotlib-2.2.3-py37hb69df0a_0
  - defaults::matplotlib-2.2.2-py37hb69df0a_2
  - defaults::numexpr-2.6.5-py37hedc7406_0
  - defaults::numexpr-2.6.8-py37h9e4a6bb_0
  - defaults::numexpr-2.6.9-py37h9e4a6bb_0
  - conda-forge::numexpr-2.6.9-py37hf8a1672_0
  - conda-forge::numexpr-2.6.8-py37hf8a1672_0
  - defaults::numexpr-2.6.5-py37h6cfcb94_0
  - defaults::numexpr-2.6.6-py37h98f89e9_0
  - defaults::pytest-doctestplus-0.2.0-py37_0
  - defaults::pytest-doctestplus-0.3.0-py37_0
  - defaults::pytest-doctestplus-0.1.3-py37_0
  - conda-forge::libcblas-3.8.0-8_openblas
  - conda-forge::numba-0.40.0-py37h637b7d7_1000
  - defaults::numba-0.43.0-py37h962f231_0
  - defaults::numba-0.42.0-py37h962f231_0
  - defaults::numba-0.40.0-py37h962f231_0
  - conda-forge::numba-0.45.0-py37hb3f55d8_0
  - conda-forge::numba-0.41.0-py37hf8a1672_0
  - conda-forge::numba-0.40.1-py37h637b7d7_1000
  - conda-forge::mkl_fft-1.0.13-py37h516909a_1
  - defaults::mkl_fft-1.0.4-py37h4414c95_1
  - conda-forge::mkl_fft-1.0.12-py37h14c3975_0
  - defaults::mkl_fft-1.0.2-py37h651fb7a_0
  - conda-forge::mkl_fft-1.0.11-py37h14c3975_0
  - conda-forge::mkl_fft-1.0.12-py37h516909a_1
  - conda-forge::mkl_fft-1.0.10-py37h14c3975_1
  - conda-forge::h5py-2.8.0-py37he5c79e1_5
  - conda-forge::h5py-2.9.0-mpi_mpich_hc9c12dd_1001
  - conda-forge::h5py-2.9.0-nompi_hf008753_1101
  - conda-forge::h5py-2.8.0-py37hfb8609e_1004
  - conda-forge::h5py-2.9.0-mpi_mpich_py37h94f87e1_1003
  - defaults::h5py-2.8.0-py37h39dcb92_0
  - conda-forge::h5py-2.9.0-nompi_h79c4800_1101
  - defaults::pandas-0.23.2-py37h04863e7_0
  - conda-forge::pandas-0.25.0-py37hb3f55d8_0
  - defaults::pandas-0.24.2-py37he6710b0_0
  - defaults::pandas-0.23.3-py37h04863e7_0
  - conda-forge::pandas-0.23.4-py37h637b7d7_1000
  - defaults::pandas-0.23.4-py37h04863e7_0
  - conda-forge::pandas-0.24.2-py37hb3f55d8_0
  - defaults::scikit-learn-0.19.2-py37h22eb022_0
  - defaults::scikit-learn-0.20.0-py37h22eb022_0
  - defaults::scikit-learn-0.20.3-py37hd81dba3_0
  - defaults::scikit-learn-0.19.2-py37h4989274_0
  - defaults::scikit-learn-0.21.1-py37h22eb022_0
  - conda-forge::scikit-learn-0.20.2-py37_blas_openblashebff5e3_1400
  - defaults::scikit-learn-0.20.0-py37h22eb022_1
  - conda-forge::pywavelets-1.0.3-py37hd352d35_0
  - conda-forge::pywavelets-1.0.1-py37h7eb728f_0
  - conda-forge::pywavelets-1.0.2-py37h3010b51_0
  - defaults::pywavelets-1.0.3-py37hdd07704_1
  - conda-forge::pywavelets-1.0.3-py37hd352d35_1
  - defaults::pywavelets-1.0.2-py37hdd07704_0
  - defaults::pywavelets-1.0.0-py37hdd07704_0
  - conda-forge::astropy-3.0.5-py37h470a237_0
  - conda-forge::astropy-2.0.14-py37h14c3975_0
  - conda-forge::astropy-3.0.4-py37h470a237_0
  - conda-forge::astropy-3.1.1-py37h470a237_0
  - defaults::astropy-2.0.13-py37h7b6447c_0
  - defaults::astropy-3.0.5-py37h7b6447c_0
  - conda-forge::astropy-3.0.4-py37h14c3975_1000
  - defaults::bokeh-1.3.1-py37_0
  - conda-forge::bokeh-1.0.4-py37_1000
  - conda-forge::bokeh-1.0.0-py37_1000
  - conda-forge::bokeh-1.0.1-py37_1000
  - conda-forge::bokeh-1.1.0-py37_0
  - defaults::bokeh-1.0.2-py37_0
  - defaults::bokeh-1.3.2-py37_0
  - conda-forge::dask-1.2.2-py_1
  - conda-forge::dask-0.17.0-py_0
  - conda-forge::dask-0.17.5-py_0
  - conda-forge::dask-2.1.0-py_0
  - defaults::dask-0.18.1-py37_0
  - defaults::dask-0.19.0-py37_0
  - defaults::dask-2.2.0-py_0
  - conda-forge::liblapack-3.8.0-4_h6e990d7_netlib
  - conda-forge::liblapack-3.8.0-6_h6e990d7_netlib
  - defaults::numpy-base-1.15.1-py37h2f8d375_0
  - defaults::numpy-base-1.9.3-py37h2b20989_5
  - defaults::numpy-base-1.9.3-py37hc937464_7
  - defaults::numpy-base-1.16.2-py37h2f8d375_0
  - defaults::nomkl-3.0-0
  - defaults::openblas-0.3.6-0
  - defaults::openblas-0.2.20-4
  - defaults::scipy-1.2.1-py37he2b7bc3_0
  - defaults::scipy-1.2.0-py37h7c811a0_0
  - conda-forge::scipy-1.3.0-py37h921218d_1
  - defaults::scipy-1.3.0-py37he2b7bc3_0
  - defaults::scipy-1.1.0-py37he2b7bc3_2
  - defaults::scipy-1.1.0-py37hfa4b5c9_1
  - defaults::scipy-1.1.0-py37h9b217d5_1
  - conda-forge::matplotlib-base-3.1.0-py37hfd891ef_1
  - conda-forge::matplotlib-base-2.2.3-py36h60b886d_1
  - conda-forge::matplotlib-base-3.1.1-py37hfd891ef_0
  - conda-forge::matplotlib-base-2.2.4-py27hfd891ef_0
  - conda-forge::matplotlib-base-3.1.0-py36hfd891ef_1
  - conda-forge::matplotlib-base-3.1.1-py36he7580a8_1
  - conda-forge::matplotlib-base-3.0.2-py36h167e16e_1002
  - conda-forge::blas-1.1-openblas
  - conda-forge::blas-1.0-openblas
  - conda-forge::blas-2.8-openblas
  - conda-forge::nltk-3.2.5-py_0
  - defaults::nltk-3.4-py37_0
  - defaults::openblas-devel-0.2.20-h1e4b922_3
  - defaults::openblas-devel-0.3.3-1
  - defaults::openblas-devel-0.3.2-1
  - defaults::openblas-devel-0.2.20-7
  - defaults::openblas-devel-0.3.3-2
  - defaults::openblas-devel-0.2.20-4
  - defaults::openblas-devel-0.3.6-1
  - conda-forge::libblas-3.8.0-8_openblas
  - conda-forge::imageio-2.4.1-py37_0
  - conda-forge::imageio-2.3.0-py_1
  - defaults::imageio-2.3.0-py37_0
  - defaults::imageio-2.4.1-py37_0
  - conda-forge::imageio-2.5.0-py37_0
  - conda-forge::imageio-2.4.1-py37_1000
  - defaults::imageio-2.5.0-py37_0
  - conda-forge::mkl-service-2.0.1-py37h516909a_0
  - defaults::mkl-service-2.0.2-py37h7b6447c_0
  - conda-forge::mkl-service-2.0.2-py37h516909a_0
  - conda-forge::seaborn-0.9.0-py_1
  - defaults::seaborn-0.9.0-py37_0
  - conda-forge::seaborn-0.9.0-py_0
  - conda-forge::seaborn-0.8.1-py_1
  - defaults::seaborn-0.8.1-py37_0
  - conda-forge::liblapacke-3.8.0-8_openblas
  - conda-forge::gensim-3.5.0-py37hd3cffdd_0
  - conda-forge::gensim-3.7.3-py37he1b5a44_0
  - conda-forge::gensim-3.7.1-py37he1b5a44_1
  - defaults::gensim-3.4.0-py37h14c3975_0
  - conda-forge::gensim-3.7.3-py37he1b5a44_1
  - conda-forge::gensim-3.5.0-py37h04863e7_1000
  - conda-forge::gensim-3.7.1-py37he1b5a44_0
  - defaults::mkl_random-1.0.2-py37hd81dba3_0
  - conda-forge::mkl_random-1.0.4-py37hf2d7682_0
  - defaults::mkl_random-1.0.1-py37h4414c95_1
  - conda-forge::mkl_random-1.0.2-py37h637b7d7_2
  - conda-forge::pytables-3.5.0-py37h5d69a80_0
  - defaults::pytables-3.5.2-py37h71ec239_1
  - conda-forge::pytables-3.4.4-py37h2392372_1
  - conda-forge::pytables-3.4.4-py37h72a6d90_4
  - conda-forge::pytables-3.5.2-py37ha1aa75f_0
  - defaults::pytables-3.4.4-py37he1c29e1_0
  - conda-forge::pytables-3.4.4-py37h2392372_3
  - defaults::pytest-arraydiff-0.2-py37h39e3cac_0
  - conda-forge::pytest-arraydiff-0.2-py_0
  - defaults::pytest-arraydiff-0.3-py37h39e3cac_0
  - conda-forge::pytest-arraydiff-0.3-py_0
  - defaults::scikit-image-0.14.1-py37he6710b0_0
  - defaults::scikit-image-0.14.2-py37he6710b0_0
  - conda-forge::scikit-image-0.14.1-py37hfc679d8_4
  - conda-forge::scikit-image-0.14.1-py37hf484d3e_1004
  - conda-forge::scikit-image-0.15.0-py37he1b5a44_0
  - conda-forge::scikit-image-0.14.2-py37hf484d3e_0
  - conda-forge::scikit-image-0.14.1-py37hfc679d8_0
  - conda-forge::statsmodels-0.10.1-py37hc1659b7_0
  - defaults::statsmodels-0.10.1-py37hdd07704_0
  - defaults::statsmodels-0.10.0-py37hdd07704_0
  - conda-forge::statsmodels-0.9.0-py37h3010b51_1000
  - defaults::statsmodels-0.9.0-py37h035aef0_0
  - conda-forge::statsmodels-0.9.0-py37h7eb728f_0
  - conda-forge::statsmodels-0.10.0-py37hc1659b7_0
  - conda-forge::pytest-astropy-0.2.1-py_0
  - conda-forge::pytest-astropy-0.4.0-py_0
  - defaults::pytest-astropy-0.5.0-py37_0
  - conda-forge::pytest-astropy-0.5.0-py_0
  - conda-forge::pytest-astropy-0.3.0-py_0
  - defaults::pytest-astropy-0.4.0-py37_0
  - defaults::anaconda-2019.03-py37_0
  - defaults::_anaconda_depends-2019.03-py37_0
  - conda-forge::patsy-0.5.0-py_1
  - defaults::patsy-0.5.0-py37_0
  - conda-forge::patsy-0.5.1-py_0
  - defaults::patsy-0.5.1-py37_0
  - defaults::bkcharts-0.2-py37_0
  - conda-forge::bottleneck-1.2.1-py37h3010b51_1001
  - conda-forge::bottleneck-1.2.1-py37h7eb728f_1
  - defaults::bottleneck-1.2.1-py37h035aef0_1

**Comment by adalisan** [comment]
For my case, I may have used --no-update-deps at one point which may have been the issue. Also I have two conda installations with some  envs  in  /home/sadali/.conda/envs  and pkgs  /home/sadali/.conda/pkgs in from a anaconda2 install

**Comment by msarahan** [comment]
When conda encounters a conflict in your specs (either within your specs, or between your specs and your environment), conda tries to generate hints to help you understand what you can do about it.  This process is pretty hard, because a graph of the packages and their matching dependencies is very costly to obtain.  It's easy to have a graph of just package names, but when you introduce versions and build numbers, it gets huge quickly.  I suspect that you are encountering this inefficiency in building the graph.  We have released conda 4.7.11 today - it will be on the conda-canary channel soon.  There was a lot of work to improve this.  It's still probably not going to be pretty, but hopefully it cuts the time down to give you a hint.  It also adds better progress bars when generating those hints, so at least you can see that it is making some progress.
